src/model/Material.hpp

```diff
@@ -95,6 +101,25 @@ namespace model {
     // resets the material property moisture penetration depth settings
     void resetMaterialPropertyMoisturePenetrationDepthSettings();
 
+    // if material property phase change already exists, do nothing and return nil; creates the material property phase change if it does not already exist and return it
+    boost::optional<MaterialPropertyPhaseChange> createMaterialPropertyPhaseChange();
+    boost::optional<MaterialPropertyPhaseChange> createMaterialPropertyPhaseChange(const std::vector<TemperatureEnthalpy>& temperatureEnthalpys);
+
+    // returns the material property phase change if set
+    boost::optional<MaterialPropertyPhaseChange> materialPropertyPhaseChange() const;
+
+    // resets the material property phase change
+    void resetMaterialPropertyPhaseChange();
+
+    // if material property phase change hysteresis already exists, do nothing and return nil; creates the material property phase change hysteresis if it does not already exist and return it
+    boost::optional<MaterialPropertyPhaseChangeHysteresis> createMaterialPropertyPhaseChangeHysteresis();
+
+    // returns the material property phase change hysteresis if set
+    boost::optional<MaterialPropertyPhaseChangeHysteresis> materialPropertyPhaseChangeHysteresis() const;
+
+    // resets the material property phase change hysteresis
+    void resetMaterialPropertyPhaseChangeHysteresis();
```

Comment:
Given the comments above, I think this is misplaced. It should be only in MasslessOpaqueMaterial and StandardOpaqueMaterial. The others shouldn't be allowed to carry one, no?

Reply:
The IDD is definitely misleading then...

Reply:
Yes it is.